shm/cargo.hpp

```diff
@@ -0,0 +1,20 @@
+#pragma once
+
+#include <string>
+
+class Cargo {
+  private: 
+     std::string name_;
+     size_t amount_;
+     size_t basePrice_;
```

Comment:
Use normal type here (int, unisigned int , uint32_t or typedef (e.g Gold))

Reply:
For amount and basePrice (and similar ones) i used uint8_t it should be enough for now

Reply:
@kasiasieradzka sorry for confusion, my usage of uint8_ was not the best idea. As @g00fer said uint32_t will be better. If you want you can change that, if not I will change it during merge

Reply:
@Kasperice, it fixed

Reply:
thanks :)